internal/auth/ldap/account_attribute_map.go

```diff
@@ -114,27 +114,41 @@ type AttributeMap struct {
 // ParseAccountAttributeMaps will parse the inbound attribute maps
 func ParseAccountAttributeMaps(ctx context.Context, m ...string) ([]AttributeMap, error) {
 	const op = "ldap.ParseAccountAttributeMaps"
-	var b kvbuilder.Builder
-	if err := b.Add(m...); err != nil {
-		return nil, errors.New(ctx, errors.InvalidParameter, op, "error parsing attribute", errors.WithWrap(err))
-	}
-	fromKeys := make([]string, 0, len(m))
-	for k := range b.Map() {
-		fromKeys = append(fromKeys, k)
-	}
-	sort.Strings(fromKeys)
-
-	attrMap := make([]AttributeMap, 0, len(fromKeys))
-	for _, from := range fromKeys {
-		var ok bool
-		to, ok := b.Map()[from].(string)
-		if !ok {
-			return nil, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("account attribute map %s value %q is not a string", from, b.Map()[from]))
+
+	am := make([]AttributeMap, 0, len(m))
+	for _, s := range m {
+		// Split into key/value which maps From/To
+		parts := strings.SplitN(s, "=", 2)
+		if len(parts) != 2 {
+			return nil, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("error parsing attribute map %q: format must be key=value", s))
+		}
+		from, to := parts[0], parts[1]
+		toAttr, err := ConvertToAccountToAttribute(ctx, to)
+		if err != nil {
+			return nil, errors.Wrap(ctx, err, op)
 		}
-		attrMap = append(attrMap, AttributeMap{
-			To:   to,
+		found := containsFunc(am, func(m AttributeMap) bool {
+			if m.To == to {
+				return true
+			}
+			return false
+		})
+		if found {
+			return nil, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("duplicate map for %q attribute", toAttr))
+		}
+		am = append(am, AttributeMap{
+			To:   string(to),
 			From: from,
 		})
 	}
-	return attrMap, nil
+	sort.Slice(am, func(i, j int) bool {
+		return am[i].From < am[j].From
+	})
+	return am, nil
+}
+
+// containsFunc reports whether at least one
+// element e of s satisfies f(e).
+func containsFunc[E any](s []E, f func(E) bool) bool {
```

Comment:
This function exists in `slices`: https://pkg.go.dev/golang.org/x/exp/slices#ContainsFunc

Reply:
🤔 this isn't part of the standard library, you should be able to just `go get golang.org/x/exp/slices` to get this function.

Reply:
it's part of the stdlib, just not in 1.19.1
my guess is that it was added with 1.20

Reply:
But the package you're importing has this function, all I'm saying is, update the package version and use the one in the package instead of reimplementing it here. It doesn't matter if `slices` will eventually be part of the standard library, right?

Reply:
sorry.. you're correct.  updating the pkg